samza-core/src/main/java/org/apache/samza/util/ProcessorIdUtil.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.util;
+
+import org.apache.samza.config.ApplicationConfig;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigException;
+import org.apache.samza.runtime.ProcessorIdGenerator;
+
+public class ProcessorIdUtil {
```

Comment:
+1 for @prateekm 's suggestion. 
Creating a public util class is dangerous as it is now open for any component to use it. This will additionally impose a requirement on the implementing class of the interface that multiple calls to generateProcessorId should return the same identifier. This is a much stronger constraint on the interface. Also, if the generator is class-loaded each time, then it cannot uphold the contract. 

Let's make sure that ProcessorIdGenerator is used *only* by the JobCoordinator. Thanks!